--- a/compat.h
+++ b/compat.h
@@ -3,12 +3,24 @@
 #define _COMPAT_H
 
 #include <linux/version.h>
+#include <linux/netdevice.h>
 
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 10, 0)
 #include <linux/vmalloc.h>
 #define compat_const const
 #else
 #define compat_const
 #endif
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 18, 0)
+static inline void compat_init_dummy_netdev(struct net_device *dev)
+{
+	dev->reg_state = NETREG_DUMMY;
+	set_bit(__LINK_STATE_PRESENT, &dev->state);
+	set_bit(__LINK_STATE_START, &dev->state);
+}
+
+#define init_dummy_netdev compat_init_dummy_netdev
+#endif
+
 #endif /* _COMPAT_H */
