--- a/gre/nss_connmgr_gre.c
+++ b/gre/nss_connmgr_gre.c
@@ -1709,31 +1709,31 @@ uint32_t nss_connmgr_gre_get_hlen(struct
  *	Map User configured flags to tunnel specific flag.
  */
 void nss_connmgr_gre_set_gre_flags(struct nss_connmgr_gre_cfg *cfg,
-					   uint16_t *o_flags, uint16_t *i_flags)
+					   unsigned long *o_flags, unsigned long *i_flags)
 
 {
 	if (cfg->ikey_valid) {
-		*i_flags |= TUNNEL_KEY;
+		set_bit(IP_TUNNEL_KEY_BIT, i_flags);
 	}
 
 	if (cfg->okey_valid) {
-		*o_flags |= TUNNEL_KEY;
+		set_bit(IP_TUNNEL_KEY_BIT, o_flags);
 	}
 
 	if (cfg->iseq_valid) {
-		*i_flags |= TUNNEL_SEQ;
+		set_bit(IP_TUNNEL_SEQ_BIT, i_flags);
 	}
 
 	if (cfg->oseq_valid) {
-		*o_flags |= TUNNEL_SEQ;
+		set_bit(IP_TUNNEL_SEQ_BIT, o_flags);
 	}
 
 	if (cfg->icsum_valid) {
-		*i_flags |= TUNNEL_CSUM;
+		set_bit(IP_TUNNEL_CSUM_BIT, i_flags);
 	}
 
 	if (cfg->ocsum_valid) {
-		*o_flags |= TUNNEL_CSUM;
+		set_bit(IP_TUNNEL_CSUM_BIT, o_flags);
 	}
 }
 
@@ -1741,33 +1741,33 @@ void nss_connmgr_gre_set_gre_flags(struc
  * nss_connmgr_gre_get_nss_config_flags()
  *	Map tunnel specific flags to NSS flags.
  */
-uint32_t nss_connmgr_gre_get_nss_config_flags(uint16_t o_flags, uint16_t i_flags,
+uint32_t nss_connmgr_gre_get_nss_config_flags(const unsigned long *o_flags, const unsigned long *i_flags,
 						  uint8_t tos, uint8_t ttl,
 						  uint16_t frag_off)
 {
 	uint32_t gre_flags = 0;
 
-	if (i_flags & TUNNEL_KEY) {
+	if (test_bit(IP_TUNNEL_KEY_BIT, i_flags)) {
 		gre_flags |= NSS_GRE_CONFIG_IKEY_VALID;
 	}
 
-	if (o_flags & TUNNEL_KEY) {
+	if (test_bit(IP_TUNNEL_KEY_BIT, o_flags)) {
 		gre_flags |= NSS_GRE_CONFIG_OKEY_VALID;
 	}
 
-	if (i_flags & TUNNEL_SEQ) {
+	if (test_bit(IP_TUNNEL_SEQ_BIT, i_flags)) {
 		gre_flags |= NSS_GRE_CONFIG_ISEQ_VALID;
 	}
 
-	if (o_flags & TUNNEL_SEQ) {
+	if (test_bit(IP_TUNNEL_SEQ_BIT, o_flags)) {
 		gre_flags |= NSS_GRE_CONFIG_OSEQ_VALID;
 	}
 
-	if (i_flags & TUNNEL_CSUM) {
+	if (test_bit(IP_TUNNEL_CSUM_BIT, i_flags)) {
 		gre_flags |= NSS_GRE_CONFIG_ICSUM_VALID;
 	}
 
-	if (o_flags & TUNNEL_CSUM) {
+	if (test_bit(IP_TUNNEL_CSUM_BIT, o_flags)) {
 		gre_flags |= NSS_GRE_CONFIG_OCSUM_VALID;
 	}
 
--- a/gre/nss_connmgr_gre.h
+++ b/gre/nss_connmgr_gre.h
@@ -23,6 +23,7 @@
 #ifndef _NSS_CONNMGR_GRE_H_
 #define _NSS_CONNMGR_GRE_H_
 
+#include <net/ip_tunnels.h>
 #include "nss_connmgr_gre_public.h"
 
 /*
@@ -81,9 +82,9 @@ typedef struct nss_gre_info nss_connmgr_
 
 uint32_t nss_connmgr_gre_get_hlen(struct nss_connmgr_gre_cfg *cfg);
 void nss_connmgr_gre_set_gre_flags(struct nss_connmgr_gre_cfg *cfg,
-					   uint16_t *o_flags, uint16_t *i_flags);
+					   unsigned long *o_flags, unsigned long *i_flags);
 
-uint32_t nss_connmgr_gre_get_nss_config_flags(uint16_t o_flags, uint16_t i_flags,
+uint32_t nss_connmgr_gre_get_nss_config_flags(const unsigned long *o_flags, const unsigned long *i_flags,
 						  uint8_t tos, uint8_t ttl,
 						  uint16_t frag_off);
 int nss_connmgr_gre_set_wifi_next_hop(struct net_device *wifi_vdev);
