--- a/nss_qdisc/nss_qdisc_stats.c
+++ b/nss_qdisc/nss_qdisc_stats.c
@@ -83,7 +83,7 @@ static void nss_qdisc_stats_sync_restart
 #if (LINUX_VERSION_CODE < KERNEL_VERSION(4, 15, 0))
 	struct nss_qdisc *nq = (struct nss_qdisc *)data;
 #else
-	struct nss_qdisc_stats_wq *nqsw = from_timer(nqsw, tm, stats_get_timer);
+	struct nss_qdisc_stats_wq *nqsw = timer_container_of(nqsw, tm, stats_get_timer);
 	struct nss_qdisc *nq = nqsw->nq;
 #endif
 
@@ -491,7 +491,7 @@ void nss_qdisc_stats_sync_many_exit(stru
 	 * We will have to wait until we have received the pending response.
 	 */
 cleanup:
-	if (del_timer(&nqsw->stats_get_timer) > 0) {
+	if (timer_delete(&nqsw->stats_get_timer) > 0) {
 		/*
 		 * The timer was still active (counting down) when it was deleted.
 		 * Therefore we are sure that there are no pending stats request
