--- a/netlink/nss_nludp_st.c
+++ b/netlink/nss_nludp_st.c
@@ -164,7 +164,7 @@ static void nss_nludp_st_time_sync_timer
 #if (LINUX_VERSION_CODE < KERNEL_VERSION(4, 15, 0))
 	struct nss_nludp_st_private *priv = (struct nss_nludp_st_private *)data;
 #else
-	struct nss_nludp_st_private *priv = from_timer(priv, t, time_sync_timer);
+	struct nss_nludp_st_private *priv = timer_container_of(priv, t, time_sync_timer);
 #endif
 	/*
 	 * Check if stop is called or test duration is over.
@@ -198,7 +198,7 @@ static void nss_nludp_st_test_duration_t
 #if (LINUX_VERSION_CODE < KERNEL_VERSION(4, 15, 0))
 	struct nss_nludp_st_private *priv = (struct nss_nludp_st_private *)data;
 #else
-	struct nss_nludp_st_private *priv = from_timer(priv, t, test_duration_timer);
+	struct nss_nludp_st_private *priv = timer_container_of(priv, t, test_duration_timer);
 #endif
 	atomic_set(&priv->disable_time_sync_timer, 1);
 }
@@ -235,7 +235,7 @@ static void nss_nludp_st_process_resp_ti
 			 * Update time sync timer disable flag and delete test duration timer.
 			 */
 			atomic_set(&global.disable_time_sync_timer, 1);
-			del_timer(&global.test_duration_timer);
+			timer_delete(&global.test_duration_timer);
 			break;
 
 		case NSS_UDP_ST_RESET_STATS_MSG:
