--- a/exports/nss_ipv4_reasm.h
+++ b/exports/nss_ipv4_reasm.h
@@ -53,7 +53,25 @@ struct nss_ipv4_reasm_stats_notification
 
 #ifdef __KERNEL__ /* only kernel will use. */
 
+/**
+ * nss_ipv4_reasm_get_context
+ *	Gets the NSS context instance for IPv4 reassembly.
+ *
+ * @return
+ * NSS context instance for IPv4 reassembly.
+ */
+extern struct nss_ctx_instance *nss_ipv4_reasm_get_context(void);
+
+/**
+ * nss_ipv4_reasm_register_handler
+ *	Registers the IPv4 reassembly message handler.
+ *
+ * @return
+ * None.
+ */
+extern void nss_ipv4_reasm_register_handler(void);
+
 /**
  * nss_ipv4_reasm_stats_register_notifier
  *	Registers a statistics notifier.

--- a/exports/nss_ipv6_reasm.h
+++ b/exports/nss_ipv6_reasm.h
@@ -57,5 +57,20 @@ struct nss_ipv6_reasm_stats_notification
 #ifdef __KERNEL__
 
 /**
+ * nss_ipv6_reasm_get_context
+ *	Gets the NSS context instance for IPv6 reassembly.
+ *
+ * @return
+ * NSS context instance for IPv6 reassembly.
+ */
+extern struct nss_ctx_instance *nss_ipv6_reasm_get_context(void);
+
+/**
+ * nss_ipv6_reasm_register_handler
+ *	Registers the IPv6 reassembly message handler.
+ */
+extern void nss_ipv6_reasm_register_handler(void);
+
+/**
  * nss_ipv6_reasm_stats_register_notifier
  *	Registers a statistics notifier.

--- a/nss_ipv4_reasm_strings.c
+++ b/nss_ipv4_reasm_strings.c
@@ -20,6 +20,7 @@
 #include "nss_core.h"
 #include <nss_ipv4_reasm.h>
 #include "nss_strings.h"
+#include "nss_ipv4_reasm_strings.h"
 
 /*
  * nss_ipv4_reasm_strings_stats

--- a/nss_ipv6_reasm_strings.c
+++ b/nss_ipv6_reasm_strings.c
@@ -20,6 +20,7 @@
 #include "nss_core.h"
 #include <nss_ipv6_reasm.h>
 #include "nss_strings.h"
+#include "nss_ipv6_reasm_strings.h"
 
 /*
  * nss_ipv6_reasm_strings_stats
