--- a/ecm_interface.c
+++ b/ecm_interface.c
@@ -4057,7 +4057,7 @@ static uint32_t ecm_interface_multicast_
 					 * For MLO bond netdevice, destination for multicast is bond netdevice itself
 					 * Therefore, slave lookup is not needed.
 					 */
-				if (ecm_front_end_is_lag_master(dest_dev) && !bond_is_mlo_device(dest_dev)) {
+				if (ecm_front_end_is_lag_master(dest_dev)) {
 					/*
 					 * Link aggregation
 					 * Figure out which slave device of the link aggregation will be used to reach the destination.
